serviceregistration/kubernetes/client/client.go

```diff
@@ -0,0 +1,282 @@
+package client
+
+import (
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/cenkalti/backoff"
+	"github.com/hashicorp/go-cleanhttp"
+	"github.com/hashicorp/go-hclog"
+)
+
+// maxRetries is the maximum number of times the client
+// should retry.
+const maxRetries = 10
+
+var (
+	ErrNamespaceUnset = errors.New(`"namespace" is unset`)
+	ErrPodNameUnset   = errors.New(`"podName" is unset`)
+	ErrNotFound       = errors.New("not found")
```

Comment:
I think this error could be improved for the user.  If you specify a bogus pod name or namespace, this is the error:

`Error running service_registration of type kubernetes: not found`

Reply:
Totally, I will fix that, great find.

Reply:
Addressed in https://github.com/hashicorp/vault/pull/8249/commits/c211b87183fb94eec61b37f8c486202ef355ce52